src/modelfree/shaping_wrappers.py

```diff
@@ -0,0 +1,208 @@
+        :param noise_annealer: Annealer.get_value - presumably the noise should be decreased
+        over time in order to get the adversarial policy to perform well on a normal victim.
+        :param noise_type: str - the type of noise parametrized by noise_annealer's value.
+        Current options are [gaussian]
+        """
+        self.agent = agent
+        self.sess = agent.sess
+        self.noise_annealer = noise_annealer
+        self.noise_generator = self._get_noise_generator(noise_type)
+
+    def _get_noise_generator(self, noise_type):
+        noise_generators = {
+            'gaussian': lambda x, size: np.random.normal(scale=x, size=size)
+        }
+        return noise_generators[noise_type]
+
+    def predict(self, observation, state=None, mask=None, deterministic=False):
+        original_actions, states = self.agent.predict(observation, state, mask, deterministic)
+        action_shape = original_actions.shape
+
+        noise_param = self.noise_annealer()
+        noise = self.noise_generator(noise_param, action_shape)
+        normed_noise = original_actions * noise
+
+        noisy_actions = original_actions + normed_noise
+        return noisy_actions, states
+
+    def reset(self):
+        return self.agent.reset()
+
+    def setup_model(self):
+        pass
+
+    def learn(self):
+        raise NotImplementedError()
+
+    def action_probability(self, observation, state=None, mask=None, actions=None):
+        raise NotImplementedError()
+
+    def save(self, save_path):
+        raise NotImplementedError()
+
+    def load(self):
+        raise NotImplementedError()
+
+
+class HumanoidEnvWrapper(VecEnvWrapper):
+    def __init__(self, env):
+        super().__init__(env)
+        self.env = env
+
+    def reset(self):
+        return self.env.reset()
+
+    def step_wait(self):
+        obs, rew, done, infos = self.env.step_wait()
+        num_envs = len(obs)
+        for env_num in range(num_envs):
+            rew[env_num] -= infos[env_num]['reward_linvel']
+
+        return obs, rew, done, infos
+
+
+def apply_env_wrapper(single_env, rew_shape_params, env_wrapper_type, scheduler):
+    if rew_shape_params is not None:
+        shaping_params = RewardShapingEnv.default_shaping_params
+        if rew_shape_params is not None:
+            config = json.load(open(rew_shape_params))
+            shaping_params.update(config)
+
+        rew_shape_anneal_frac = shaping_params.get('rew_shape_anneal_frac', 0)
+        assert 0 <= rew_shape_anneal_frac <= 1
+        if rew_shape_anneal_frac > 0:
+            rew_shape_func = LinearAnnealer(1, 0, rew_shape_anneal_frac).get_value
+        else:
+            rew_shape_func = None
+
+        scheduler.set_func('rew_shape', rew_shape_func)
+        return RewardShapingEnv(single_env, shaping_params=shaping_params,
+                                reward_annealer=scheduler.get_func('rew_shape'))
+    else:
+        wrappers = {
+            'humanoid': HumanoidEnvWrapper,
```

Comment:
Might be unnecessary if we just make this a new environment (see earlier comment).